test/livebook_teams/hubs/team_client_test.exs

```diff
@@ -698,11 +714,24 @@ defmodule Livebook.Hubs.TeamClientTest do
                         sessions: [%{app_status: %{execution: :executed}}]
                       }}
 
+      assert_receive {:apps_manager_status, [%{app_spec: ^app_spec, running?: true}]}
       assert app_deployment in TeamClient.get_app_deployments(team.id)
 
-      agent_connected = %{agent_connected | app_deployments: []}
-      send(pid, {:event, :agent_connected, agent_connected})
+      # TODO: Replace this with a better solution
+      Process.sleep(100)
```

Comment:
Perhaps you could start a process that subscribes to particular teams topic and proxies messages, like this:

```elixir
parent = self()

Node.spawn_link(node, fn ->
  # pubsub subscribe
  
  loop = fn loop ->
    receive do
      message ->
        send(parent, {:teams_broadcast, message})
        loop.(loop)
    end
  end

  loop.(loop)
end)

assert_receive {:teams_broadcast, ...}
```

You can avoid the funky anonymous function, if you encapsulate the logic in the helper module on the teams side.

Then we await on the status update to arrive.